File: backend-nodejs/functions-ingestion/src/providers/azure/searchStore.ts
import type { SearchStoreProvider } from "../base.js";
import {
  indexChunkDocuments,
  searchChunkDocuments,
  deleteSearchChunksForDocument,
  countSearchChunksForDocument,
  listSearchDocumentGroups,
  searchEnabled
} from "../../shared/searchIndexStore.js";

export class AzureSearchStoreProvider implements SearchStoreProvider {
  isEnabled(): boolean {
    return searchEnabled();
  }

  async indexChunks(chunks: Record<string, unknown>[]): Promise<boolean> {
    return indexChunkDocuments(
      chunks as Parameters<typeof indexChunkDocuments>[0]
    );
  }

  async searchChunks(
    query: string,
    tenantId: string,
    top = 3,
    queryEmbedding?: number[],
    mode?: string
  ): Promise<Record<string, unknown>[]> {
    const results = await searchChunkDocuments(
      query,
      tenantId,
      top,
      queryEmbedding,
      mode as Parameters<typeof searchChunkDocuments>[4]
    );
    return results as Record<string, unknown>[];
  }

  async deleteChunksForDocument(
    documentId: string,
    tenantId: string
  ): Promise<number> {
    return deleteSearchChunksForDocument(documentId, tenantId);
  }

  async countChunksForDocument(
    documentId: string,
    tenantId: string
  ): Promise<number> {
    return countSearchChunksForDocument(documentId, tenantId);
  }

  async listDocumentGroups(
    tenantId: string
  ): Promise<Record<string, unknown>[]> {
    const results = await listSearchDocumentGroups(tenantId);
    return results as Record<string, unknown>[];
  }
}

File: backend-python/app/providers/azure/search_store.py
from __future__ import annotations

from typing import Any, Dict, List

from ..base import SearchStoreProvider
from ...search import (
    delete_search_chunks_for_document,
    list_search_document_groups,
    upsert_search_chunks,
)
from ...config import search_enabled


class AzureSearchStoreProvider(SearchStoreProvider):
    def upsert_chunks(
        self,
        tenant_id: str,
        document_id: str,
        blob_name: str,
        file_name: str,
        chunks: List[str],
    ) -> None:
        upsert_search_chunks(tenant_id, document_id, blob_name, file_name, chunks)

    def delete_chunks_for_document(self, document_id: str, tenant_id: str) -> int:
        return delete_search_chunks_for_document(document_id, tenant_id)

    def list_document_groups(self, tenant_id: str) -> List[Dict[str, Any]]:
        return list_search_document_groups(tenant_id)

    def is_configured(self) -> bool:
        return search_enabled()
